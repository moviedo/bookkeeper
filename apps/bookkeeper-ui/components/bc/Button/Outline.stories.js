import { Default as ButtonDefault } from "./Default.stories";

export default {
  title: "base components / bc-button / outlined",
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes).filter((a) => a !== "default"),
  template: `
  <bc-button v-bind="$props">
    ${args.default}
  </bc-button>
  `,
});

export const Default = Template.bind({});
Default.args = {
  ...ButtonDefault.args,
  outline: true,
};

export const Hover = Template.bind({});
Hover.args = { ...Default.args };
Hover.parameters = {
  pseudo: {
    hover: true,
  },
};

export const Focus = Template.bind({});
Focus.args = { ...Default.args };
Focus.parameters = {
  pseudo: {
    focus: true,
  },
};

export const Disabled = Template.bind({});
Disabled.args = {
  ...Default.args,
  disabled: true,
};
