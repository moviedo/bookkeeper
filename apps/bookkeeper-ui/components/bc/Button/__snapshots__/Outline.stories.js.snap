// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base components / bc-button / outlined Default smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
        outline="true"
>
  Submit
</button>
`;

exports[`base components / bc-button / outlined Disabled smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
        outline="true"
        disabled="disabled"
>
  Submit
</button>
`;

exports[`base components / bc-button / outlined Focus smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
        outline="true"
>
  Submit
</button>
`;

exports[`base components / bc-button / outlined Hover smoke-test 2`] = `
<button data-v-85cb8f92
        class="default hover focus active"
        outline="true"
>
  Submit
</button>
`;
